nilearn/plotting/surf_plotting.py

```diff
@@ -305,6 +307,153 @@ def plot_surf(surf_mesh, surf_map=None, bg_map=None,
         return figure
 
 
+def _get_faces_on_edge(faces, parc_idx):
+    '''Internal function for identifying which faces lie on the outer edge of the parcellation defined by the indices in parc_idx.
+    IN:
+    faces      -   numpy ndarray of shape (n, 3), containing indices of the mesh faces
+    parc_idx   -   indices of the vertices belonging to the region that is to be plotted
+    OUT:
+    boolean array indicating which faces lie on the outer edge
+    '''
+    # count how many vertices belong to the given parcellation in each face
+    verts_per_face = np.array([np.isin(face, parc_idx) for face in faces]).sum(axis=1)
+
+    # test if parcellation forms regions
+    if np.all(verts_per_face < 2):
+        raise ValueError('Vertices in parcellation do not form region.')
+
+    vertices_on_edge = np.intersect1d(np.unique(faces[verts_per_face == 2]),
+                                      parc_idx)
+    faces_outside_edge = np.sum(
+        [np.isin(face, vertices_on_edge) for face in faces], axis=1)
+
+    return np.logical_and(faces_outside_edge > 0, verts_per_face < 3)
+
+
+def plot_surf_contours(surf_mesh, roi_map, axes=None, figure=None, levels=None,
+                       labels=None, colors=None, legend=False, cmap='tab20',
+                       title=None, output_file=None, **kwargs):
+    """ Plotting contours of regions of interest on a surface, possibly overlaying a statistical map.
+
+    Parameters
+    ----------
+    surf_mesh: str or list of two numpy.ndarray
+        Surface mesh geometry, can be a file (valid formats are
+        .gii or Freesurfer specific files such as .orig, .pial,
+        .sphere, .white, .inflated) or
+        a list of two Numpy arrays, the first containing the x-y-z coordinates
+        of the mesh vertices, the second containing the indices
+        (into coords) of the mesh faces.
+
+    roi_map : str or numpy.ndarray or list of numpy.ndarray
+        ROI map to be displayed on the surface mesh, can be a file
+        (valid formats are .gii, .mgz, .nii, .nii.gz, or Freesurfer specific
+        files such as .annot or .label), or
+        a Numpy array with a value for each vertex of the surf_mesh.
+        The value at each vertex one inside the ROI and zero inside ROI, or an
+        integer giving the label number for atlases.
+
+    axes: instance of matplotlib axes, None, optional
+        The axes instance to plot to. The projection must be '3d' (e.g.,
+        `figure, axes = plt.subplots(subplot_kw={'projection': '3d'})`,
+        where axes should be passed.).
+        If None, axes instance from figure is used if available,else a new axes is created.
+
+    figure: instance of matplotlib figure, None, optional
+        The figure instance to plot to. If None, parent figure of axes is used if available, else a new figure is created.
```

Comment:
Beware the 80 characters limit. You may want to configure pep8 or flake8 on your editor.

Reply:
If you do that, make sure it only modifies lines that you have touched to avoid huge diffs of formatting changes that aren't related to your PR. or have it just highlight problematic lines without changing them